src/nvim/eval.c

```diff
@@ -8018,26 +8019,47 @@ static void f_bufnr(typval_T *argvars, typval_T *rettv)
     rettv->vval.v_number = -1;
 }
 
-/*
- * "bufwinnr(nr)" function
- */
-static void f_bufwinnr(typval_T *argvars, typval_T *rettv)
+static void buf_win_common(typval_T *argvars, typval_T *rettv, bool get_nr)
 {
-  (void)get_tv_number(&argvars[0]);         /* issue errmsg if type error */
-  ++emsg_off;
+  int error = false;
+  (void)get_tv_number_chk(&argvars[0], &error);  // issue errmsg if type error
```

Comment:
This does not detect `v:true` as an error/invalid argument.

`get_buf_tv()` only cares about `Number` and `String`, everything else is an error (`NULL`). No conversion from `v:true` to `1` is made.  Thats why `{expr} = v:true` is not the same as `{expr} = 1` and should be an error.